cmake/adjust/template/vendor/CXX_Clang.cmake

```diff
@@ -0,0 +1,51 @@
+include_guard(GLOBAL)
+
+# Debug
+set(NBL_CXX_DEBUG_COMPILE_OPTIONS
+	-ggdb3 -Wall -fno-omit-frame-pointer -fstack-protector-strong
```

Comment:
is `ggdb3` sufficient?

Reply:
shouldn't `-g` also be used?

Reply:
btw GCC/Clang have an equivalent of ffast math that we use in MSVC

Reply:
actually `ggdb3` doesn't exist on clang and it throws a warning about unknown arg

Reply:
also ggdb3 is for GDB, I think on Clang one is supposed to use LLDB

Reply:
on windows `ClangCL` which is Clang frontend compatible with MSVC CLI uses `lld-link` linker

> C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\bin>lld.exe
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead